dialogflow/stopwatch/README.md

```diff
@@ -1,42 +1,70 @@
 # Dialogflow Sample
 
-This app demonstrates how to make streaming gRPC connections to the [Dialogflow API](https://cloud.google.com/dialogflow-enterprise/) to recognize commands in recorded audio.
+This app demonstrates how to make gRPC connections to the [Dialogflow API](https://cloud.google.com/dialogflow-enterprise/)
+
+The app demonstrates how to detect intents:
+- Via Text
+- Via Streaming Audio
+- With Sentiment Analysis
+- With Text-to-Speech
+- With Knowledge Connectors
+
+To call the Dialogflow API from iOS, you need to provide authorization tokens with each request for them to be accepted by the Dialogflow API. To get this token, this sample uses a Firebase Function (in Node.js) to generate these tokens on the behalf of a service account to be used by the app when making a request to the Dialogflow API.
 
 ## Prerequisites
-- Credentials for calling the Dialogflow API
-- An OSX machine or emulator
+- An OSX machine or simulator
 - [Xcode 9.1][xcode] or later
 - [Cocoapods][cocoapods] version 1.0 or later
 
-## Prepare your Google Cloud account
-As with all Google Cloud APIs, every call to the Dialogflow API must be associated
-  with a project within the [Google Cloud Console][cloud-console] that has the
-  API enabled. In brief:
-  - Create a project (or use an existing one) in the [Cloud Console][cloud-console]
-  - [Enable billing][billing].
+## Setup
+- Create a project with the [Google Cloud Console][cloud-console]
+- Enable the [Dialogflow API](https://console.cloud.google.com/apis/librarydialogflow.googleapis.com).
+- [Enable billing][billing].
+- [Import the Dialogflow Agent](https://dialogflow.com/docs/agents/export-import-restore#import) using the `StopwatchAgent.zip` which is located in the `stopwatch` directory. 
+- [Create a Service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) with the following IAM role: Dialogflow API Client. Example name: `dialogflow-client`
+- For your "App Engine Default Service Account" add the following role "Service Account Token Creator" Role.
+- To use [Sentiment Analysis] you need to [Enable beta features](https://cloud.google.com/dialogflow-enterprise/docs/sentiment#enable_beta_features)
+- To use [Text-to-Speech] you need to [Enable beta features](https://cloud.google.com/dialogflow-enterprise/docs/detect-intent-tts#enable_beta_features)
+- To use [Knowledge Connectors] you need to [Enable beta features](https://cloud.google.com/dialogflow-enterprise/docs/knowledge-connectors#enable_beta_features)
```

Comment:
Can remove the `[]` brackets surrounding the beta features if you aren't linking to anything.

Reply:
Done.